packages/core/src/EventNormalizer.ts

```diff
@@ -581,26 +579,27 @@ export class EventNormalizer {
         // Reconstruct the inserted text from the computed indices.
         const insertedText = current.chars.slice(insertStart, insertEnd);
 
-        // Compute the range in the previous DOM corresponding to the range of
-        // the observed correction in current DOM. For this purpose, the nodes
-        // in the `previous` mapping were given a reference to their original
-        // self before cloning such that it can be retrieved now.
+        // Compute the selection in the previous DOM corresponding to the
+        // selection of the observed correction in current DOM. For this
+        // purpose, the nodes in the `previous` mapping were given a reference
+        // to their original self before cloning such that it can be retrieved
+        // now.
         const previousNodes = previous.nodes as ClonedNode[];
         // The indices in the current DOM must be offset by the difference in
         // length between the previous and current content in order to compute
         // their corresponding indices in the previous DOM.
         const insertPreviousStart = insertStart;
         const insertPreviousEnd = insertEnd + previousLength - currentLength;
-        const insertionRange: DomRangeDescription = {
-            startContainer: previousNodes[insertPreviousStart].origin,
-            startOffset: previous.offsets[insertPreviousStart],
-            endContainer: previousNodes[insertPreviousEnd].origin,
-            endOffset: previous.offsets[insertPreviousEnd],
+        const impliedSelection: DomSelectionDescription = {
```

Comment:
It is not a range and it is not *the* selection. It is implied by the mutations yes. For example, you write `Christofe` then you click on the suggestion of your spell-checking keyboard which says `Christophe`. We implement this as a `setSelection(Christofe)` followed by `insertText(Christophe)` so the selection is *implied* because you never *actually* selected anything in the text. I can change `implied` to something else if you still think the name is wrong after this explanation. Now that I think about it and the commands can receive a range as parameter, it might be more correct to trigger an insertText with this *insertion range* than fake a `setSelection` to use it as the range of the command.

Reply:
Right, that does make more sense. Seems like this kind of thing is even the whole point of this PR isn't it?